var/spack/repos/builtin/packages/wgrib2/package.py

```diff
@@ -84,62 +151,73 @@ class Wgrib2(MakefilePackage):
     variant("jasper", default=True, description="JPEG compression using Jasper")
     variant("openmp", default=True, description="OpenMP parallelization")
     variant("wmo_validation", default=False, description="WMO validation")
+    #    variant("shared", default=False, description="Enable shared library", when="+lib")
+    variant("disable_stat", default=False, description="Disable POSIX feature", when="@:3.1")
+    variant("openjpeg", default=False, description="Enable OpenJPEG")
+    variant(
+        "enable_docs", default=False, description="Build doxygen documentation", when="@3.4.0:"
+    )
 
     conflicts("+netcdf3", when="+netcdf4")
+    conflicts("+netcdf3", when="+netcdf")
     conflicts("+openmp", when="%apple-clang")
 
-    depends_on("wget", type=("build"), when="+netcdf4")
-    # makefile behavior with shell commands/character escapes breaks with gmake@4.3:
-    depends_on("gmake@:4.2")
-
-    variant_map = {
-        "netcdf3": "USE_NETCDF3",
-        "netcdf4": "USE_NETCDF4",
-        "spectral": "USE_SPECTRAL",
-        "mysql": "USE_MYSQL",
-        "udf": "USE_UDF",
-        "regex": "USE_REGEX",
-        "tigge": "USE_TIGGE",
-        "proj4": "USE_PROJ4",
-        "aec": "USE_AEC",
-        "g2c": "USE_G2CLIB",
-        "png": "USE_PNG",
-        "jasper": "USE_JASPER",
-        "openmp": "USE_OPENMP",
-        "wmo_validation": "USE_WMO_VALIDATION",
-        "ipolates": "USE_IPOLATES",
-        "disable_timezone": "DISABLE_TIMEZONE",
-        "disable_alarm": "DISABLE_ALARM",
-        "fortran_api": "MAKE_FTN_API",
-    }
-
-    # Disable parallel build
-    parallel = False
+    depends_on("ip@5.1:", when="@3.5: +ipolates")
+    depends_on("lapack", when="@3.5: +ipolates")
+    depends_on("libaec@1.0.6:", when="@3.2: +aec")
+    depends_on("netcdf-c", when="@3.2: +netcdf4")
+    depends_on("jasper@:2", when="@3.2:3.4 +jasper")
+    depends_on("g2c", when="@3.5: +jasper")
+    depends_on("zlib-api", when="@3.2: +png")
+    depends_on("libpng", when="@3.2: +png")
+    depends_on("openjpeg", when="@3.2:3.4 +openjpeg")
+    depends_on("g2c +openjpeg", when="@3.5: +openjpeg")
+    requires("^g2c@1.9:", when="@3.5: ^g2c")
+
+    @when("@:2 ^gmake@4.2:")
+    def patch(self):
+        filter_file(r"\\#define", "#define", "makefile")
 
     # Use Spack compiler wrapper flags
     def inject_flags(self, name, flags):
-        spec = self.spec
         if name == "cflags":
-            if spec.satisfies("%apple-clang"):
+            if self.spec.compiler.name == "apple-clang":
```

Comment:
@AlexanderRichert-NOAA This seems to be a bug, or at least it is causing a build error? https://github.com/JCSDA/spack-stack/actions/runs/13077825823/job/36494196074?pr=1486

Maybe the compiler name is still `clang`, not `apple-clang`, even if the spec is apple-clang?

Reply:
Did you already take care of this? I can't test it, so feel free to do whatever makes work...

Reply:
My first attempt failed, I will need to look again. Unlikely I will be able to finish this today, other things are more urgent.